snow/networking/handler/handler.go

```diff
@@ -46,11 +47,14 @@ var (
 
 	errMissingEngine  = errors.New("missing engine")
 	errNoStartingGear = errors.New("failed to select starting gear")
+	errorShuttingDown = errors.New("shutting down")
 )
 
 type Handler interface {
 	health.Checker
 
+	Notify(_ context.Context, msg common.Message) error
```

Comment:
Does `Notify` need to be added to the `Handler` interface? Now that we are creating the `NotificationForwarder` internally, I think we pass in the struct rather than an interface, so it feels like this isn't needed.